game-core/src/main/java/games/strategy/triplea/ai/AbstractAi.java

```diff
@@ -103,9 +103,7 @@ public boolean confirmMoveKamikaze() {
   }
 
   @Override
-  public boolean confirmMoveHariKari() {
-    return false;
-  }
+  public void confirmMoveHariKari() {}
```

Comment:
Here the issue becomes very clear.
From a readers perspective this looked like the Ai was denying HariKari in general.
However if you think about it confirmation of some sort only makes sense for human players, the Ai is not going to be like "ok, right I didn't notice I would lose those units".
So move this method to a different interface maybe?